app/controllers/api/v1/posts_controller.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling post requests.
+    class PostsController < ApplicationController
+      before_action :authenticate_user!, except: %i[index show]
+      before_action :set_post, except: %i[index create]
+      before_action :set_tag, only: %i[remove_tag]
+      before_action :authorize_post!
+      after_action :verify_authorized
+
+      # GET /posts or /posts.json
+      def index
+        render json: Post.all, status: :ok
```

Comment:
(for later improvement) 
TODO: 
in order to filter posts by tag, we may pass tag as an argument